game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleState.java

```diff
@@ -23,6 +24,8 @@ public Side getOpposite() {
 
   Territory getBattleSite();
 
+  UUID getBattleId();
```

Comment:
I can do that but I'd still have to cast it back to a UUID where it is called because the code requires it to be a UUID.

Reply:
Also, the method already exists and is public on AbstractBattle.  This is just adding the method to the interface.  So I'd either have to make a new method or change a lot of code to make `getBattleId` return a `BattleId` or `String`.

Reply:
Is there a serialization concern as well? It does sound like coming back and changing this is the right thing to do. Casting back to a UUID is part of the reason why it's brittle as-is.

Reply:
Yes, it is stored on `AbstractBattle` so serialization in saves would be a problem.  I glanced at a few places where it is used and they also look like they are saved.  And I think it might also be used for client/server communication so it is serialized there as well.

Reply:
That sounds right, when switching from GUID to UUID, there were serialization implications. Well :hankey: